### DISCUSSION TITLE
qpdf

### DISCUSSION THREAD

**Discussion by m-holger** [m-holger/pdfarranger#12] [General]
@jberkenbilt 

> I am probably going to send something out to the qpdf-announce list at some point to discuss a future API for doing page-level operations. Are you subscribed?

I was not aware this list exists. I am now subscribed.

> I would welcome your input

Happy to contribute if I can.

I am afraid I have spend the last 30+ years doing my best to avoid looking at any flavour of C code, so I feel at times unsure of my interpretation of your code. Would you be prepared to confirm from time to time my understanding of your code (I'll try to phrase things to allow yes/no answers) or point me at the right piece of code to look at? 

Thanks again for your help, and for qpdf.

**Comment by jberkenbilt** [comment]
@m-holger No problem. Happy to have as little or as much of your input as appropriate, and happy to answer questions. I'm quite sure James Barlow will provide input since pikepdf is one of the more sophisticated and complete users of the qpdf API. He's already created many of his own higher-level interfaces, and I've made a few changes to qpdf to make it easier for pikepdf. Even if some of your input were just things you wish you could do more easily, exposing functionality in the qpdf API makes it easier for it to be exposed in pikepdf.
